Group component tests failing when pytest run in parallel

<!-- READ THIS FIRST:
  - If you need additional help with this template, please refer to https://www.home-assistant.io/help/reporting_issues/
  - Make sure you are running the latest version of Home Assistant before reporting an issue: https://github.com/home-assistant/core/releases
  - Do not report issues for integrations if you are using custom components or integrations.
  - Provide as many details as possible. Paste logs, configuration samples and code into the backticks.
  DO NOT DELETE ANY TEXT from this template! Otherwise, your issue may be closed without comment.
-->
## The problem
<!-- 
  Describe the issue you are experiencing here to communicate to the
  maintainers. Tell us what you were trying to do and what happened.
-->

When I run `pytest -n auto tests/components/group` on my 8700k (6C/12T) I consistently get these tests that fail:

```
FAILED tests/components/group/test_init.py::test_group_that_references_a_group_of_covers[pyloop]
FAILED tests/components/group/test_init.py::test_group_that_references_two_groups_of_covers[pyloop]
FAILED tests/components/group/test_init.py::test_group_that_references_two_types_of_groups[pyloop]
FAILED tests/components/group/test_init.py::test_device_tracker_not_home[pyloop]
FAILED tests/components/group/test_init.py::test_cover_added_after_group[pyloop]
```

I've created one issue for all of them, since they fail the same way.

I'd really like to use make use of pytest-xdist, since it makes developing a much more delightful experience.

## Environment
<!--
  Provide details about the versions you are using, which helps us to reproduce
  and find the issue quicker. Version information is found in the
  Home Assistant frontend: Configuration -> Info.
-->

```
============================= test session starts ==============================
platform linux -- Python 3.8.5, pytest-5.4.3, py-1.9.0, pluggy-0.13.1
rootdir: /build/source, inifile: setup.cfg, testpaths: tests
plugins: aiohttp-0.3.0, xdist-1.34.0, forked-1.3.0, requests-mock-1.8.0
gw0 [1331] / gw1 [1331] / gw2 [1331] / gw3 [1331] / gw4 [1331] / gw5 [1331] / gw6 [1331] / gw7 [1331] / gw8 [1331] / gw9 [1331] / gw10 [1331] / gw11 [1331]
```

- Home Assistant Core release with the issue:  0.116.0
- Last working Home Assistant Core release (if known):  unknown
- Operating environment (OS/Container/Supervised/Core):  NixOS
- Integration causing this issue: group
- Link to integration documentation on our website: https://www.home-assistant.io/integrations/group/

## Problem-relevant `configuration.yaml`
<!--
  An example configuration that caused the problem for you. Fill this out even
  if it seems unimportant to you. Please be sure to remove personal information
  like passwords, private URLs and other credentials.
-->

n/a

## Traceback/Error logs
<!--
  If you come across any trace or error logs, please provide them.
-->

```txt
_____________ test_group_that_references_a_group_of_covers[pyloop] _____________
[gw5] linux -- Python 3.8.5 /nix/store/z65l1jqvxa58zzwwa3bvglb6asj4y8cv-python3-3.8.5/bin/python3.8

hass = <homeassistant.core.HomeAssistant object at 0x7fffeefec2b0>

    async def test_group_that_references_a_group_of_covers(hass):
        """Group that references a group of covers."""

        entity_ids = [
            "cover.living_front_ri",
            "cover.living_back_lef",
        ]
        hass.state = CoreState.stopped

        for entity_id in entity_ids:
            hass.states.async_set(entity_id, "closed")
        await hass.async_block_till_done()

        assert await async_setup_component(
            hass,
            "group",
            {
                "group": {
                    "living_room_downcover": {"entities": entity_ids},
                    "grouped_group": {
                        "entities": ["group.living_room_downlights", *entity_ids]
                    },
                }
            },
        )
        await hass.async_block_till_done()

        hass.bus.async_fire(EVENT_HOMEASSISTANT_START)
        await hass.async_block_till_done()

>       assert hass.states.get("group.living_room_downcover").state == "closed"
E       AssertionError: assert 'unknown' == 'closed'
E         - closed
E         + unknown

tests/components/group/test_init.py:1133: AssertionError
```

```txt
___________ test_group_that_references_two_groups_of_covers[pyloop] ____________
[gw5] linux -- Python 3.8.5 /nix/store/z65l1jqvxa58zzwwa3bvglb6asj4y8cv-python3-3.8.5/bin/python3.8

hass = <homeassistant.core.HomeAssistant object at 0x7fffef2a4520>

    async def test_group_that_references_two_groups_of_covers(hass):
        """Group that references a group of covers."""

        entity_ids = [
            "cover.living_front_ri",
            "cover.living_back_lef",
        ]
        hass.state = CoreState.stopped

        for entity_id in entity_ids:
            hass.states.async_set(entity_id, "closed")
        await hass.async_block_till_done()

        assert await async_setup_component(
            hass,
            "group",
            {
                "group": {
                    "living_room_downcover": {"entities": entity_ids},
                    "living_room_upcover": {"entities": entity_ids},
                    "grouped_group": {
                        "entities": [
                            "group.living_room_downlights",
                            "group.living_room_upcover",
                        ]
                    },
                }
            },
        )
        await hass.async_block_till_done()

        hass.bus.async_fire(EVENT_HOMEASSISTANT_START)
        await hass.async_block_till_done()

>       assert hass.states.get("group.living_room_downcover").state == "closed"
E       AssertionError: assert 'unknown' == 'closed'
E         - closed
E         + unknown

tests/components/group/test_init.py:1171: AssertionError
```

```txt
____________ test_group_that_references_two_types_of_groups[pyloop] ____________
[gw5] linux -- Python 3.8.5 /nix/store/z65l1jqvxa58zzwwa3bvglb6asj4y8cv-python3-3.8.5/bin/python3.8

hass = <homeassistant.core.HomeAssistant object at 0x7fffeefecb80>

    async def test_group_that_references_two_types_of_groups(hass):
        """Group that references a group of covers and device_trackers."""

        group_1_entity_ids = [
            "cover.living_front_ri",
            "cover.living_back_lef",
        ]
        group_2_entity_ids = [
            "device_tracker.living_front_ri",
            "device_tracker.living_back_lef",
        ]
        hass.state = CoreState.stopped

        for entity_id in group_1_entity_ids:
            hass.states.async_set(entity_id, "closed")
        for entity_id in group_2_entity_ids:
            hass.states.async_set(entity_id, "home")
        await hass.async_block_till_done()

        assert await async_setup_component(
            hass,
            "group",
            {
                "group": {
                    "covers": {"entities": group_1_entity_ids},
                    "device_trackers": {"entities": group_2_entity_ids},
                    "grouped_group": {
                        "entities": ["group.covers", "group.device_trackers"]
                    },
                }
            },
        )
        await hass.async_block_till_done()

        hass.bus.async_fire(EVENT_HOMEASSISTANT_START)
        await hass.async_block_till_done()

>       assert hass.states.get("group.covers").state == "closed"
E       AssertionError: assert 'unknown' == 'closed'
E         - closed
E         + unknown

tests/components/group/test_init.py:1213: AssertionError
```

```txt
_____________________ test_device_tracker_not_home[pyloop] _____________________
[gw6] linux -- Python 3.8.5 /nix/store/z65l1jqvxa58zzwwa3bvglb6asj4y8cv-python3-3.8.5/bin/python3.8

hass = <homeassistant.core.HomeAssistant object at 0x7fffef0a5d30>

    async def test_device_tracker_not_home(hass):
        """Test group of device_tracker not_home."""
        hass.states.async_set("device_tracker.one", "not_home")
        hass.states.async_set("device_tracker.two", "not_home")
        hass.states.async_set("device_tracker.three", "not_home")

        assert await async_setup_component(
            hass,
            "group",
            {
                "group": {
                    "group_zero": {
                        "entities": "device_tracker.one, device_tracker.two, device_tracker.three"
                    },
                }
            },
        )
        await hass.async_block_till_done()

>       assert hass.states.get("group.group_zero").state == "not_home"
E       AssertionError: assert 'unknown' == 'not_home'
E         - not_home
E         + unknown

tests/components/group/test_init.py:914: AssertionError
```

```txt
_____________________ test_cover_added_after_group[pyloop] _____________________
[gw5] linux -- Python 3.8.5 /nix/store/z65l1jqvxa58zzwwa3bvglb6asj4y8cv-python3-3.8.5/bin/python3.8

hass = <homeassistant.core.HomeAssistant object at 0x7fffef6b35b0>

    async def test_cover_added_after_group(hass):
        """Test cover added after group."""

        entity_ids = [
            "cover.upstairs",
            "cover.downstairs",
        ]

        assert await async_setup_component(
            hass,
            "group",
            {
                "group": {
                    "shades": {"entities": entity_ids},
                }
            },
        )
        await hass.async_block_till_done()

        for entity_id in entity_ids:
            hass.states.async_set(entity_id, "open")
        await hass.async_block_till_done()
        await hass.async_block_till_done()

>       assert hass.states.get("group.shades").state == "open"
E       AssertionError: assert 'unknown' == 'open'
E         - open
E         + unknown

tests/components/group/test_init.py:1060: AssertionError
```

## Additional information

--- a/tests/components/group/test_init.py
+++ b/tests/components/group/test_init.py
@@ -763,6 +763,7 @@ async def test_group_climate_all_cool(hass):
     hass.states.async_set("climate.two", "cool")
     hass.states.async_set("climate.three", "cool")
 
+    assert await async_setup_component(hass, "climate", {})
     assert await async_setup_component(
         hass,
         "group",
@@ -772,7 +773,6 @@ async def test_group_climate_all_cool(hass):
             }
         },
     )
-    assert await async_setup_component(hass, "climate", {})
     await hass.async_block_till_done()
 
     assert hass.states.get("group.group_zero").state == STATE_ON
@@ -1041,6 +1041,7 @@ async def test_cover_added_after_group(hass):
         "cover.downstairs",
     ]
 
+    assert await async_setup_component(hass, "cover", {})
     assert await async_setup_component(
         hass,
         "group",
@@ -1125,6 +1126,8 @@ async def test_group_that_references_a_group_of_covers(hass):
             }
         },
     )
+
+    assert await async_setup_component(hass, "cover", {})
     await hass.async_block_till_done()
 
     hass.bus.async_fire(EVENT_HOMEASSISTANT_START)
@@ -1147,6 +1150,7 @@ async def test_group_that_references_two_groups_of_covers(hass):
         hass.states.async_set(entity_id, "closed")
     await hass.async_block_till_done()
 
+    assert await async_setup_component(hass, "cover", {})
     assert await async_setup_component(
         hass,
         "group",
@@ -1192,6 +1196,7 @@ async def test_group_that_references_two_types_of_groups(hass):
         hass.states.async_set(entity_id, "home")
     await hass.async_block_till_done()
 
+    assert await async_setup_component(hass, "device_tracker", {})
     assert await async_setup_component(
         hass,
         "group",
@@ -1205,6 +1210,7 @@ async def test_group_that_references_two_types_of_groups(hass):
             }
         },
     )
+    assert await async_setup_component(hass, "cover", {})
     await hass.async_block_till_done()
 
     hass.bus.async_fire(EVENT_HOMEASSISTANT_START)

